bundled/tool/server.py

```diff
@@ -84,6 +85,35 @@ def did_save(params: lsp.DidSaveTextDocumentParams) -> None:
     diagnostics: list[lsp.Diagnostic] = _linting_helper(document)
     LSP_SERVER.publish_diagnostics(document.uri, diagnostics)
 
+NOQA_REGEX = re.compile(r"(?i:# noqa)(?::\s?(?P<codes>([A-Z]+[0-9]+(?:[,\s]+)?)+))?")
+CODE_REGEX = re.compile(r"[A-Z]+[0-9]+")
+
+
+@LSP_SERVER.feature(lsp.HOVER)
+def hover(params: lsp.HoverParams) -> None:
```

Comment:
Another option to show a check summary is via Diagnostics.


https://user-images.githubusercontent.com/17039389/202880911-fafb038d-9ac5-45dc-b790-8946d40a52cc.mov

In this demo, I tweaked the ruff source code to show check code locations as violations.